tests/e2e/cli-upgrade/assert-odigos-upgraded.yaml

```diff
@@ -0,0 +1,158 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: odigos-test-cli-upgrade
+  labels:
+    odigos.io/system-object: "true"
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app.kubernetes.io/name: odigos-autoscaler
+  namespace: odigos-test-cli-upgrade
+spec:
+  containers:
+    - name: manager
+      image: keyval/odigos-autoscaler:e2e-test
```

Comment:
review note: i tested it with an illegal tag value to make sure this is asserted correctly